### ISSUE TITLE
Spaces around operator check: false positives when captured functions present

### ISSUE THREAD

**Issue by lauraannwilliams** [rrrene/credo#370]
### Precheck
Posting as per request in #298

### Environment

* Credo version (`mix credo -v`): 0.7.3
* Erlang/Elixir version (`elixir -v`): 
Erlang/OTP 19 [erts-8.3] [source] [64-bit] [smp:8:8] [async-threads:10] [hipe] [kernel-poll:false] [dtrace]

Elixir 1.4.2

* Operating system: macOs Sierra 10.12.4

### What were you trying to do?
I'm writing up an absinthe schema, which involves a lot of captured functions. I noticed that when I put in a <> operator I got a spaces around operator check credo notice.

After some further investigation, I found the warning does not show up running the standard `mix credo` command, but did show up in atom, where I'm using [linter-elixir-credo](https://github.com/smeevil/linter-elixir-credo)

A little investigation showed that the linter is using the ```--read-from-stdin``` flag. When I use that flag I can reproduce the behaviour:

### Expected outcome

No credo Warnings

### Actual outcome

```
cat credo_test.ex  | mix credo --format=flycheck --read-from-stdin credo_test.ex
credo_test.ex:12:11: C: There are no spaces around operators most of the time, but here there are.
credo_test.ex:11:11: C: There are no spaces around operators most of the time, but here there are.
```

A minimal file to reproduce this error is

```
defmodule CredoTest do
  @moduledoc """
  minimal file to reproduce unexpected "spaces around operators most of the time" error
  when the number of function captures is equal to or greater than the number
  of other operator uses, the warning is triggered. 
  """

  def test do

    &String.capitalize/1
    &String.downcase/1
    &String.reverse/1

    "foo" <> " Add more."
    1 + 1 = 2

  end
end
```

**Comment by rrrene** [comment]
> After some further investigation

Ha! Excellent work. I added a test to reproduce this and will fix it when I'm home. Thx :+1:
